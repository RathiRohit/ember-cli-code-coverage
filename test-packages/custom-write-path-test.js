'use strict';

const fs = require('fs-extra');
const util = require('util');
const rimraf = util.promisify(require('rimraf'));
const { dir, file } = require('chai-files');
const path = require('path');
const execa = require('execa');

const BASE_PATH = path.join(__dirname, 'my-app-with-custom-write-path');

describe('app coverage generation with custom writePath', function () {
  jest.setTimeout(10000000);

  beforeEach(async function () {
    await rimraf(`${BASE_PATH}/coverage*`);
    await execa('git', ['clean', '-f', 'my-app-with-custom-write-path'], { cwd: __dirname });
    await execa('git', ['restore', 'my-app-with-custom-write-path'], { cwd: __dirname });
  });

  afterEach(async function () {
    await rimraf(`${BASE_PATH}/coverage*`);
    await execa('git', ['clean', '-f', 'my-app-with-custom-write-path'], { cwd: __dirname });
    await execa('git', ['restore', 'my-app-with-custom-write-path'], { cwd: __dirname });
  });

  it('runs coverage with custom writePath set', async function () {
    dir(`${BASE_PATH}/coverage`).assertDoesNotExist();

    let env = { COVERAGE: 'true' };
    await execa('ember', ['test'], { cwd: BASE_PATH, env });
    file(`${BASE_PATH}/coverage/lcov-report/index.html`).assertIsNotEmpty();
    file(`${BASE_PATH}/coverage/index.html`).assertIsNotEmpty();

    let summary = fs.readJSONSync(`${BASE_PATH}/coverage/coverage-summary.json`);
    expect(summary).toMatchSnapshot();
  });
});
