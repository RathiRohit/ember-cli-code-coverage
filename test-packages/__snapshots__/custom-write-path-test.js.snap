// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app coverage generation with custom writePath runs coverage with custom writePath set 1`] = `
Object {
  "app/app.js": Object {
    "branches": Object {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": Object {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "lines": Object {
      "covered": 4,
      "pct": 100,
      "skipped": 0,
      "total": 4,
    },
    "statements": Object {
      "covered": 4,
      "pct": 100,
      "skipped": 0,
      "total": 4,
    },
  },
  "app/router.js": Object {
    "branches": Object {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": Object {
      "covered": 1,
      "pct": 33.33,
      "skipped": 0,
      "total": 3,
    },
    "statements": Object {
      "covered": 1,
      "pct": 33.33,
      "skipped": 0,
      "total": 3,
    },
  },
  "app/utils/my-covered-util.js": Object {
    "branches": Object {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": Object {
      "covered": 1,
      "pct": 100,
      "skipped": 0,
      "total": 1,
    },
    "lines": Object {
      "covered": 1,
      "pct": 100,
      "skipped": 0,
      "total": 1,
    },
    "statements": Object {
      "covered": 1,
      "pct": 100,
      "skipped": 0,
      "total": 1,
    },
  },
  "app/utils/my-uncovered-util.js": Object {
    "branches": Object {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "statements": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
  },
  "total": Object {
    "branches": Object {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": Object {
      "covered": 1,
      "pct": 33.33,
      "skipped": 0,
      "total": 3,
    },
    "lines": Object {
      "covered": 6,
      "pct": 66.67,
      "skipped": 0,
      "total": 9,
    },
    "statements": Object {
      "covered": 6,
      "pct": 66.67,
      "skipped": 0,
      "total": 9,
    },
  },
}
`;
